packages/percy-storybook/src/getStories.js

```diff
@@ -1,111 +1,70 @@
-import jsdom from 'jsdom';
+const puppeteer = require('puppeteer');
 import { storiesKey } from './constants';
 
-// jsdom doesn't support Web Workers yet.
-// We use workerMock to allow the user's preview.js to interact with the Worker API.
-const workerMock = `
-    function MockWorker(path) {
-      var api = this;
-
-      function addEventListener() {}
-      function removeEventListener() {}
-      function postMessage() {}
-      function terminate() {}
-
-      api.postMessage = postMessage;
-      api.addEventListener = addEventListener;
-      api.removeEventListener = removeEventListener;
-      api.terminate = terminate;
+// The function below needs to be in a template string to prevent babel from transforming it.
+// If babel transformed it, puppeteer wouldn't be able to evaluate it properly.
+// See: https://github.com/GoogleChrome/puppeteer/issues/1665#issuecomment-354241717
+// Also see comment below about Debugging page.evaluate
+const fetchStoriesFromWindow = `(async () => {
+  return await new Promise((resolve, reject) => {
+    const storiesKey = '${storiesKey}';
+    // Check if the window has stories every 100ms for up to 10 seconds.
+    // This allows 10 seconds for any async pre-tasks (like fetch) to complete.
+    // Usually stories will be found on the first loop.
+    var checkStories = function(timesCalled) {
+      if (window[storiesKey]) {
+        // Found the stories, return them.
+        resolve(window[storiesKey]);
+      } else if (timesCalled < 100) {
+        // Stories not found yet, try again 100ms from now
+        setTimeout(() => {
+          checkStories(timesCalled + 1);
+        }, 100);
+      } else {
+        reject(new Error(
+          'Stories not found on window within 10 seconds. ' +
+          "Check your call to serializeStories in your Storybook's config.js."
+        ));
+      }
+    };
+    checkStories(0);
+  });
+})()`;
 
-      return api;
-    }
-    window.Worker = MockWorker;
-`;
+export default async function getStories(options = {}) {
+  let launchArgs = [];
 
-// jsdom doesn't support localStorage yet.
-// We use localStorageMock to allow the user's preview.js to interact with localStorage.
-const localStorageMock = `
-    var localStorageMock = (function() {
-      var store = {};
-      return {
-        getItem: function(key) {
-          return store[key];
-        },
-        setItem: function(key, value) {
-          store[key] = value.toString();
-        },
-        clear: function() {
-          store = {};
-        }
-      };
-    })();
-    Object.defineProperty(window, 'localStorage', { value: localStorageMock });
-    `;
+  // Travis CI requires Chrome to be launched without the sandbox
```

Comment:
I this only true for Travis or other CI systems as well? In the latter, it would probably be good to check `process.env.CI` which a lot of different vendors set to `true`.

Reply:
@joscha I'm going to update this PR to make --no-sandbox the default.